src/parser/mod.rs

```diff
@@ -7506,7 +7506,7 @@ impl<'a> Parser<'a> {
                 with_offset,
                 with_offset_alias,
             })
-        } else if self.parse_keyword(Keyword::JSON_TABLE) {
+        } else if dialect_of!(self is MySqlDialect | AnsiDialect) && self.parse_keyword(Keyword::JSON_TABLE) {
```

Comment:
I still don't think the dialect identity should be hardcoded. This should probably be a dialect.supports_json_table. And it should be true in the generic dialect.

Reply:
I agree it would be nicer to do if `self.dialect.supports_json_table()` instead. However, I think we could also do this as a follow on PR as well

Reply:
I don't see why it should be true in generic dialect. `GenericDialect` is not strictly SQL standard dialect (`AnsiDialect` is for the purpose). It is for syntax that most dialects use. Currently the table function is only supported by very few dialects.

Reply:
> I agree it would be nicer to do if `self.dialect.supports_json_table()` instead. However, I think we could also do this as a follow on PR as well

Agreed. As I mentioned early, this can be discussed separately. Although I wonder if it will result in many similar functions.

Reply:
> Although I wonder if it will result in many similar functions.

Yes I think it would. The codebase has a mix of styles currently -- both `dialect_of!` and trait methods